src/components/shared/UserAvatar/UserAvatar.tsx

```diff
@@ -1,129 +1,111 @@
 }: Props) => {
-  // const username = getUsername(user);
-  // const username = 'tempUser';
-  // let popoverProps: InfoPopoverProps = {
-  //   popperOptions,
-  //   trigger: showInfo ? 'click' : 'disabled',
-  //   user,
-  //   showArrow: popperOptions && popperOptions.showArrow,
-  // };
-  // if ('colony' in rest) {
-  //   const { colony, domainId } = rest;
-  //   popoverProps = {
-  //     ...popoverProps,
-  //     banned,
-  //     colony,
-  //     domainId,
-  //   };
-  // }
-  // const avatar = (
-  //   <InfoPopover {...popoverProps}>
-  //     <div
-  //       // @ts-ignore
-  //       className={getMainClasses({}, styles, {
-  //         showOnClick: popoverProps.trigger === 'click',
-  //       })}
-  //     >
-  //       <Avatar
-  //         avatarURL={avatarURL}
-  //         className={className}
-  //         notSet={typeof notSet === 'undefined' ? true : notSet}
-  //         placeholderIcon="circle-person"
-  //         seed={address && address.toLowerCase()}
-  //         size={size}
-  //         title={showInfo ? '' : username || address}
-  //       />
-  //     </div>
-  //   </InfoPopover>
-  // );
-  // if (showLink && username) {
-  //   // Won't this always be lowercase?
-  //   return <Link to={`/user/${username.toLowerCase()}`}>{avatar}</Link>;
-  // }
-  // return avatar;
-  return (
-    <Avatar
-      avatarURL={avatarURL}
-      className={className}
-      notSet={typeof notSet === 'undefined' ? true : notSet}
-      placeholderIcon="circle-person"
-      seed={address && address.toLowerCase()}
-      size={size}
-      title={
-        showInfo ? '' : user?.profile?.displayName || user?.name || address
-      }
-    />
+  const trigger = showInfo ? 'click' : 'disabled';
+  const showArrow = popperOptions && popperOptions.showArrow;
+  const { profile } = user || {};
+  const imageString = preferThumbnail ? profile?.thumbnail : profile?.avatar;
+
+  const renderContent = useMemo(() => {
+    return <UserInfoPopover colony={colony} user={user} banned={banned} />;
+  }, [user, banned, colony]);
+
+  const avatar = (
+    <Popover
+      renderContent={renderContent}
+      popperOptions={popperOptions}
+      trigger={trigger}
+      showArrow={showArrow}
+    >
+      <div
+        className={getMainClasses(
+          {},
+          styles as unknown as { [k: string]: string },
+          {
+            showOnClick: trigger === 'click',
+          },
+        )}
+      >
+        <Avatar
+          avatarURL={avatarURL || imageString || undefined}
+          className={className}
+          notSet={typeof notSet === 'undefined' ? true : notSet}
+          placeholderIcon="circle-person"
+          seed={address && address.toLowerCase()}
+          size={size}
+          title={
+            showInfo ? '' : user?.profile?.displayName || user?.name || address
+          }
+        />
+      </div>
+    </Popover>
   );
+  if (showLink && user) {
+    // Won't this always be lowercase?
```

Comment:
No it won't

At least currently we don't have any transform that does this. But maybe we should look into this

Reply:
I included the comment from the dapp just incase it was relevant but if if a user name has the possibility of being uppercase it sounds like I should remove the comment and using toLowerCase() should work for now.